pkg/compiler/analysis.go

```diff
@@ -48,6 +48,7 @@ func (c *codegen) traverseGlobals() int {
 		}
 		emit.Instruction(c.prog.BinWriter, opcode.INITSSLOT, []byte{byte(n)})
 		c.ForEachFile(c.convertGlobals)
+		c.ForEachFile(c.convertInitFuncs)
```

Comment:
Will it work correctly for subpackages especially wrt ordering of these init invocations?

Reply:
No, I because `ForEachFile` traverses packages via a map. This is a problem which can also occur with globals. Shouldn't be hard to fix, will do in this PR.

Reply:
Fixed.